### ISSUE TITLE
Debounce strange effect

### ISSUE THREAD

**Issue by ftaiolivista** [cyclejs/cyclejs#939]
I was experiencing a strange behavior using debounce EXTRA.

DOM.select('mysel').events('input').debug('PRE').compose(debounce(450)).debug('AFTER')

In console I read

```
PRE: InputEvent {ownerTarget: input.mdc-text-field__input, currentTarget: input.mdc-text-field__input…}
AFTER: InputEvent {ownerTarget: div#editCncDialog.mdc-dialog.mdc-dialog--open, currentTarget: div#editCncDialog.mdc-dialog.mdc-dialog--open...}
```

How can ownerTarget change after debounce?!

N.B. Only one event is emitted. No other console output.

**Comment by jvanbruegge** [comment]
I think owner target might be mutated during bubbling, I'd have to check that

**Comment by jvanbruegge** [comment]
@staltz and I decided that this will be a "won't fix". The reason is that it would force us to allocate an object for every DOM node that needs to be traversed, leading to a big performance impact. See the memory graphs here:
![photo1628502973](https://user-images.githubusercontent.com/1529052/129340775-d355cbbd-9d71-4a8f-a7f1-eeed898a4805.jpeg)
![photo1628502997](https://user-images.githubusercontent.com/1529052/129340777-8a41a21e-7851-4e2d-a057-d960fe1f7efa.jpeg)

So if you want to use events async, you can do the shallow copy yourself like this:
```diff
DOM.select('mysel').events('input')
    .debug('PRE')
+    .map(cloneEvent)
    .compose(debounce(450))
    .debug('AFTER')

+function cloneEvent(ev: any) {
+    const newEvent = { ...ev };
+    newEvent.__proto__ = ev.__proto__;
+    return newEvent;
+}
```
```
